src/Visualization/Python/PlotDatFile.py

```diff
@@ -0,0 +1,296 @@
+                 label=this_label,
+                 linestyle=this_linestyle,
+                 linewidth=args['linewidth'])
+
+    if args['y_logscale']:
+        plt.yscale('log')
+    plt.xlabel(args['x_label'] if args['x_label'] != None else args['x_axis'],
+               fontsize=args['fontsize'])
+
+    _ = plt.legend(fontsize=args['fontsize'] - 1,
+                   loc='best',
+                   ncol=args['legend_ncols'])
+    plt.xticks(fontsize=args['fontsize'])
+    plt.yticks(fontsize=args['fontsize'])
+    if args['x_bounds']:
+        plt.xlim(float(args['x_bounds'][0]), float(args['x_bounds'][1]))
+    if args['y_bounds']:
+        plt.ylim(float(args['y_bounds'][0]), float(args['y_bounds'][1]))
+    if args['title'] != None:
+        plt.title(args['title'], fontsize=args['fontsize'])
+    plt.savefig(args['output'] + '.pdf',
+                transparent=True,
+                format='pdf',
+                bbox_inches='tight')
+    return None
+
+
+def process_subfile(dat_file, legend_only, write_dat, **kwargs):
+    """Given the read in subfile, plot the data.
+
+    If --legend-only was specified then the legend is printed
+    and execution terminated.
+
+    If --write-dat is specified then the data is written to a text dat file.
+
+    Generates a plot and writes it to disk using read_and_plot_data.
+    """
+    legend = [(x.decode('ascii') if type(x) == bytes else x)
+              for x in dat_file.attrs['Legend']]
+    if legend_only:
+        print("The legend in the dat subfile is:\n{}".format(legend))
+        return legend
+
+    if not (write_dat is None):
+        print("Writing to text dat file '{}'".format(write_dat))
+        np.savetxt(write_dat,
+                   np.asarray(dat_file),
+                   delimiter=' ',
+                   header='\n'.join(
+                       '{}' for _ in range(len(legend))).format(*legend))
+        return None
+
+    read_and_plot_data(dat_file, kwargs)
+    return None
+
+
+def open_and_process_file(filename, subfile_name, **kwargs):
+    """Opens the specified HDF5 file.
+
+    If no subfile is specified this function prints all available Dat
+    subfiles in the root of the HDF5 and then exists. If a subfile is
+    specified, the function forwards on to process_subfile()
+    """
+    def available_subfiles(h5file):
+        return [member for member in h5file.keys() if member.endswith('.dat')]
+
+    with h5py.File(filename) as h5file:
+        if subfile_name != None:
+            dat_file = h5file.get(subfile_name + '.dat')
+            if dat_file is None:
+                raise Exception(
+                    "Unable to open dat file '%s'. Available files are:\n%s" %
+                    (subfile_name + '.dat', available_subfiles(h5file)))
+            return process_subfile(dat_file, **kwargs)
+        else:
+            print("Available dat files are:", available_subfiles(h5file))
+            return available_subfiles(h5file)
```

Comment:
My experience with dealing with string encoding in python has been terrible... Even python 3 made several changes between versions. I'd prefer to just ignore it here since python 2 is EOL and while not pretty it's not a functional deficit. I have no objection to the change as a whole, just not sure it's worth the effort.

Reply:
I'm ok with leaving as is. Yeah it was mostly a cosmetic change, but I agree all the necessary info is there.